bgpd/bgp_rd.c

```diff
@@ -143,22 +141,22 @@ int str2prefix_rd(const char *str, struct prefix_rd *prd)
 			stream_putl(s, atol(p + 1));
 		}
 	} else {
-		ret = inet_aton(half, &addr);
-		if (!ret)
+		if (!inet_aton(half, &addr))
 			goto out;
 
+		s = stream_new(RD_BYTES);
```

Comment:
I am not really sure what the goal here is.  Having 2 stream_new's is far less desirable in my mind then delaying the allocation until after all the checks have been made.  I would not have blinked if the stream_new was moved to line 128, but as it is this complicates what is effectively simple code for in this case some dubious improvements to ordering.  It is always preferable for code to be simple to read especially in code that is not in any way performance critical

Reply:
Ok, i have re-pushed.

Reply:
Let's be clear on what path needs to be optimized.  I would be willing to bet that this function pretty much never fails at all.  So let's optimize for the *normal* case not the special case.

Reply:
and having said that, string parsing to read a RD is *never* going to be in a code path that needs very high performance.  In that case it is more important for code clarity then any optimizations at all